eng/pipelines/coreclr/superpmi-collect.yml

```diff
@@ -127,6 +130,7 @@ jobs:
     platforms:
     # Linux tests are built on the OSX machines.
     # - OSX_x64
+    - OSX_arm64
```

Comment:
Could you also try if it works for benchmark runs collection? Linux arm/arm64 don't run because of a known problem, but perhaps OSX_arm64 might not have that.

Reply:
Sure, I will try.

Reply:
Tried enabling the benchmarks and the job is failing at `dotnet build` stage:
```
  Determining projects to restore...
  Restored /private/tmp/helix/working/A94808F2/w/A57B0905/u/performance/src/tools/Reporting/Reporting/Reporting.csproj (in 5.33 sec).
  Restored /private/tmp/helix/working/A94808F2/w/A57B0905/u/performance/src/harness/BenchmarkDotNet.Extensions/BenchmarkDotNet.Extensions.csproj (in 5.72 sec).
  Restored /tmp/helix/working/A94808F2/w/A57B0905/u/performance/src/benchmarks/micro/MicroBenchmarks.csproj (in 5.99 sec).

Running: /tmp/helix/working/A94808F2/w/A57B0905/u/performance/tools/dotnet/arm64/dotnet build /tmp/helix/working/A94808F2/w/A57B0905/u/performance/src/benchmarks/micro/MicroBenchmarks.csproj --configuration Release --framework net7.0 --no-restore /p:NuGetPackageRoot=/tmp/helix/working/A94808F2/w/A57B0905/u/performance/artifacts/packages -o /tmp/helix/working/A94808F2/w/A57B0905/u/performance/artifacts
Microsoft (R) Build Engine version 17.2.0-preview-22060-17+14efa06d7 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Reporting -> /tmp/helix/working/A94808F2/w/A57B0905/u/performance/artifacts/Reporting.dll
  BenchmarkDotNet.Extensions -> /tmp/helix/working/A94808F2/w/A57B0905/u/performance/artifacts/BenchmarkDotNet.Extensions.dll
/private/tmp/helix/working/A94808F2/w/A57B0905/u/performance/src/benchmarks/micro/libraries/Common/Perf.TextBase.cs(5,23): error CS0234: The type or namespace name 'Attributes' does not exist in the namespace 'BenchmarkDotNet' (are you missing an assembly reference?) [/tmp/helix/working/A94808F2/w/A57B0905/u/performance/src/benchmarks/micro/MicroBenchmarks.csproj]
```

Reply:
I will need to build dotnet/performance locally on an M1 device.
For now, let's keep the job disabled.

Reply:
Can you add a TODO or something for benchmarks run?

Reply:
Sure, what is weird I don't see the issue while building the performance repo locally.
Trying to dig more

Reply:
I believe there is a fundamental issue with `dotnet restore` when running on a project path that contains symbolic link - I just finished creating a repro and submitted https://github.com/dotnet/sdk/issues/23430. 

And since `/tmp` on macOS is a symbolic link to `/private/tmp`
```
echesako@Egors-Mac-mini ~> stat -F /tmp
lrwxr-xr-x 1 root wheel 11 Dec  7 15:39:27 2021 /tmp@ -> private/tmp
```
I believe we are running into the issue.

I think I can workaround this in Python by using `os.path.realpath()`

Reply:
The workaround worked - I have a successful benchmarks collection on macOS/arm64.
Pushed the workaround and enabled macOS/arm64 in superpmi-collect job.